reco_utils/recommender/deeprec/deeprec_utils.py

```diff
@@ -0,0 +1,351 @@
+        use_DNN_part=FLAGS["use_DNN_part"] if "use_DNN_part" in FLAGS else False,
+        # train
+        init_method=FLAGS["init_method"] if "init_method" in FLAGS else "tnormal",
+        init_value=FLAGS["init_value"] if "init_value" in FLAGS else 0.01,
+        embed_l2=FLAGS["embed_l2"] if "embed_l2" in FLAGS else 0.0000,
+        embed_l1=FLAGS["embed_l1"] if "embed_l1" in FLAGS else 0.0000,
+        layer_l2=FLAGS["layer_l2"] if "layer_l2" in FLAGS else 0.0000,
+        layer_l1=FLAGS["layer_l1"] if "layer_l1" in FLAGS else 0.0000,
+        cross_l2=FLAGS["cross_l2"] if "cross_l2" in FLAGS else 0.0000,
+        cross_l1=FLAGS["cross_l1"] if "cross_l1" in FLAGS else 0.0000,
+        reg_kg=FLAGS["reg_kg"] if "reg_kg" in FLAGS else 0.0000,
+        learning_rate=FLAGS["learning_rate"] if "learning_rate" in FLAGS else 0.001,
+        lr_rs=FLAGS["lr_rs"] if "lr_rs" in FLAGS else 1,
+        lr_kg=FLAGS["lr_kg"] if "lr_kg" in FLAGS else 0.5,
+        kg_training_interval=FLAGS["kg_training_interval"]
+        if "kg_training_interval" in FLAGS
+        else 5,
+        max_grad_norm=FLAGS["max_grad_norm"] if "max_grad_norm" in FLAGS else 2,
+        is_clip_norm=FLAGS["is_clip_norm"] if "is_clip_norm" in FLAGS else 0,
+        dtype=FLAGS["dtype"] if "dtype" in FLAGS else 32,
+        loss=FLAGS["loss"] if "loss" in FLAGS else None,
+        optimizer=FLAGS["optimizer"] if "optimizer" in FLAGS else "adam",
+        epochs=FLAGS["epochs"] if "epochs" in FLAGS else 10,
+        batch_size=FLAGS["batch_size"] if "batch_size" in FLAGS else 1,
+        enable_BN=FLAGS["enable_BN"] if "enable_BN" in FLAGS else False,
+        # show info
+        show_step=FLAGS["show_step"] if "show_step" in FLAGS else 1,
+        save_model=FLAGS["save_model"] if "save_model" in FLAGS else True,
+        save_epoch=FLAGS["save_epoch"] if "save_epoch" in FLAGS else 5,
+        metrics=FLAGS["metrics"] if "metrics" in FLAGS else None,
+        write_tfevents=FLAGS["write_tfevents"] if "write_tfevents" in FLAGS else False,
+    )
+
+
+def prepare_hparams(yaml_file=None, **kwargs):
+    if yaml_file:
+        config = load_yaml(yaml_file)
+        config = flat_config(config)
+    else:
+        config = {}
+
+    if kwargs:
+        for name, value in six.iteritems(kwargs):
+            config[name] = value
+
+    check_nn_config(config)
+    hparams = create_hparams(config)
+    return hparams
+
+
+def download_deeprec_resources(azure_container_url, data_path, remote_resource_name):
+    os.makedirs(data_path, exist_ok=True)
+    remote_path = azure_container_url + remote_resource_name
+    maybe_download(remote_path, remote_resource_name, data_path)
+    zip_ref = zipfile.ZipFile(os.path.join(data_path, remote_resource_name), "r")
+    zip_ref.extractall(data_path)
+    zip_ref.close()
+    os.remove(os.path.join(data_path, remote_resource_name))
+
+
+def cal_metric(labels, preds, metrics):
```

Comment:
Just FYI, we have some utility metrics here https://github.com/Microsoft/Recommenders/blob/staging/reco_utils/evaluation/python_evaluation.py. We don't have the ones that you have here, so I guess that in the future, we will move the metrics you made to the other common location. 

This is just for your information, there's no need to make any change

Reply:
Got it. I will consider putting all evaluation metrics in one place in next version.